scripts/systemd/check_and_upgrade_serial_keel.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+# This script will either setup a fresh serial-keel service instance, or upgrade your existing one.
+# This will also be executed during every bootup, so a check for an upgrade will be performed.
+set -e
+
+CURRENT_USER=$(whoami)
+USER_HOME="/home/$CURRENT_USER/"
+
+mkdir -p $USER_HOME/.config/systemd/user/
+
+# The name of the branch can be specified as a argument, or it defaults to 'main'
+BRANCH=${1-main}
+CONFIG_PATH=${2-/home/$CURRENT_USER/config.ron}
+
+SCRIPT_PATH="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+cd $SCRIPT_PATH
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+cd $REPO_ROOT
+
+# Check if there are new changes. If there are none, and if the serial-keel service
+# already exists on the machine, then start the serial-keel server
+
+GIT_NO_BRANCH_SWITCH=true
+if [[ $(git rev-parse --abbrev-ref HEAD) != "$BRANCH" ]]; then
+    GIT_NO_BRANCH_SWITCH=false
+    git checkout $BRANCH
+fi
+
+git fetch
+
+GIT_NO_NEW_REMOTE_CHANGES=true
+if git status -uno | grep -q "Your branch is behind"; then
+    GIT_NO_NEW_REMOTE_CHANGES=false
+    git pull
+fi
+
+
+if $GIT_NO_NEW_REMOTE_CHANGES && $GIT_NO_BRANCH_SWITCH && test -f $USER_HOME/.config/systemd/user/serial-keel.service; then
+    echo "serial-keel already up to date."
+    exit 0
+fi
+
+# If not, we have a serial-keel update
+echo "We have a serial-keel update to perform.."
+mkdir -p $USER_HOME/.config/environment.d/
+env > $USER_HOME/.config/environment.d/env
```

Comment:
Perhaps surround the making of this folder and file with a test if it exists?
Since `env` is only populated when the user runs this script manually, and thereafter it would only be overwritten with the same contents by the service.